tfe/resource_tfe_workspace_run.go

```diff
@@ -0,0 +1,192 @@
+package tfe
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var (
+	backoffMin = 1000.0
+	backoffMax = 3000.0
+)
+
+var planPendingStatuses = map[tfe.RunStatus]bool{
+	tfe.RunPending:        true,
+	tfe.RunPlanQueued:     true,
+	tfe.RunPlanning:       true,
+	tfe.RunCostEstimating: true,
+	tfe.RunPolicyChecking: true,
+	tfe.RunQueuing:        true,
+	tfe.RunFetching:       true,
+}
+
+var planTerminalStatuses = map[tfe.RunStatus]bool{
+	tfe.RunPlanned:            true,
+	tfe.RunPlannedAndFinished: true,
+	tfe.RunErrored:            true,
+	tfe.RunCostEstimated:      true,
+	tfe.RunPolicyChecked:      true,
+	tfe.RunPolicySoftFailed:   true,
+	tfe.RunPolicyOverride:     true,
+}
+
+var applyPendingStatuses = map[tfe.RunStatus]bool{
+	tfe.RunConfirmed:   true,
+	tfe.RunApplyQueued: true,
+	tfe.RunApplying:    true,
+	tfe.RunQueuing:     true,
+	tfe.RunFetching:    true,
+}
+
+var applyDoneStatuses = map[tfe.RunStatus]bool{
+	tfe.RunApplied: true,
+	tfe.RunErrored: true,
+}
+
+var confirmationDoneStatuses = map[tfe.RunStatus]bool{
+	tfe.RunConfirmed:   true,
+	tfe.RunApplyQueued: true,
+	tfe.RunApplying:    true,
+}
+
+func resourceTFEWorkspaceRun() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceTFEWorkspaceRunCreate,
+		Delete:        resourceTFEWorkspaceRunDelete,
+		Read:          resourceTFEWorkspaceRunRead,
+		Update:        resourceTFEWorkspaceRunUpdate,
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"organization": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"workspace": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"apply": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
🤔 My instincts are telling me that having these two manually duplicated inner block schemas is a maintenance hazard. Do you reckon it's worth de-duplicating, with a lil helper function that allocates and returns a `map[string]*schema.Schema`?

Reply:
Sure, that makes sense to me